### DISCUSSION TITLE
Running coverage thresholds against existing reports

### DISCUSSION THREAD

**Discussion by angeloderivco** [ArtiomTr/jest-coverage-report-action#320] [Q&A]
As a background, I run my tests into separate jobs and I merge them on a separate job. I was hoping I can check my final report against the coverage threshold but not sure if it only works when you run the unit test during this action

**Comment by ArtiomTr** [comment]
Hello @angeloderivco :wave:,

you can just specify generated coverage reports, and the action will not run unit tests. Documentation for these options is [here](https://github.com/ArtiomTr/jest-coverage-report-action#use-existing-test-reports)

**Reply by angeloderivco** [reply]
Hi @ArtiomTr !

I had a read through the link you sent and the only option it seems is to check against the coverage report of the branch you're merging into, and not an actual coverage threshold similar to what you specify in Jest

**Reply by ArtiomTr** [reply]
@angeloderivco, specifying `coverage-file` and `base-coverage-file` is the same as skipping all steps, except threshold comparison & reporting. If simplified, the action does the following:

1. Install dependencies
2. Run tests
3. Checkout to the base branch
4. Install dependencies
5. Run tests
6. Compare coverage, check thresholds
7. Comment report

If you specify `coverage-file` and `base-coverage-file`, the action skips some steps:

1. ~Install dependencies~ (skipped due to the `coverage-file` option)
2. ~Run tests~ (skipped due to the `coverage-file` option)
3. ~Checkout to the base branch~ (skipped due to the `base-coverage-file` option)
4. ~Install dependencies~ (skipped due to the `base-coverage-file` option)
5. ~Run tests~ (skipped due to the `base-coverage-file` option)
6. Compare coverage, check thresholds
7. Comment report

As you can see, coverage comparison and threshold check steps *should* work the same as before specifying `coverage-file` and `base-coverage-file`. If all tests are being run through the action, does it work correctly? If not, it's possibly an issue with the threshold check algorithm.

**Reply by angeloderivco** [reply]
Thanks for the detailed answer @ArtiomTr . I'll give it a whirl and see how it goes.
